File: src/main/java/javabot/operations/TimeOperation.java
package javabot.operations;

import java.util.Calendar;

import javabot.BotEvent;
import javabot.Javabot;
import javabot.Message;

public class TimeOperation extends BotOperation {
    public TimeOperation(final Javabot javabot) {
        super(javabot);
    }

    @Override
    public boolean handleMessage(final BotEvent event) {
        final String message = event.getMessage();
        boolean handled = false;
        if ("time".equals(message) || "date".equals(message)) {
            getBot().postMessage(new Message(event.getChannel(), event, Calendar.getInstance().getTime().toString()));
            handled = true;
        }
        return handled;
    }
}

File: src/main/kotlin/javabot/operations/TimeOperation.kt
package javabot.operations

import com.google.inject.Inject
import java.util.Calendar
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao

class TimeOperation @Inject constructor(bot: Javabot, adminDao: AdminDao) :
    BotOperation(bot, adminDao) {
    override fun handleMessage(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val message = event.value
        if ("time" == message || "date" == message) {
            responses.add(Message(event, Calendar.getInstance().time.toString()))
        }
        return responses
    }
}
